### PR TITLE
Raise `IndexError` for R2C/C2R FFT with `axes=()`

### LINKED ISSUE #3240: Real FFTs with empty `axes` should fail
The shape of the output of `fft.(i)rfft*` depends on the last axis.
```
>>> numpy.fft.rfftn(numpy.arange(3), axes=())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<__array_function__ internals>", line 5, in rfftn
  File "/path/to/python3.8/site-packages/numpy-1.18.2-py3.8-linux-x86_64.egg/numpy/fft/_pocketfft.py", line 1124, in rfftn
    a = rfft(a, s[-1], axes[-1], norm)
IndexError: list index out of range
>>> cupy.fft.rfftn(cupy.arange(3), axes=())
array([0, 1, 2])
```

### PR DESCRIPTION
**Pull request by leofang**
Close #3240. This makes the behavior compatible with NumPy.

### CODE REVIEW

**Comment by grlee77** [comment]
Agreed that this matches NumPy behavior.

I'm not sure if the difference between C2C and R2C/C2R cases was intentional or an oversight on the NumPy side, though?

**Comment by leofang** [comment]
My understanding for R2C/C2R is the last axis plays a special role, whereas for C2C all axes are treated on equal footing, so if we can't define clearly which axis is the last, it'd be a problem. So it's intentional I guess.

File: cupy/fft/fft.py [hunk 1]
```diff
@@ -419,8 +422,11 @@ def _fftn(a, s, axes, norm, direction, value_type='C2C', order='A', plan=None,
                          % norm)
 
     axes, axes_sorted = _prep_fftn_axes(a.ndim, s, axes)
-    if not axes_sorted:
-        return a
+    if len(axes_sorted) == 0:
```
Comment (toslunar) [inline cupy/fft/fft.py:425]: ditto

File: cupy/fft/fft.py [hunk 2]
```diff
@@ -166,8 +166,11 @@ def _fft(a, s, axes, norm, direction, value_type='C2C', overwrite_x=False,
         axes = [i for i in range(-dim, 0)]
     else:
         axes = tuple(axes)
-    if not axes:
-        return a
+    if len(axes) == 0:
```
[the remark this answers is not in the document]
Comment (leofang) [reply cupy/fft/fft.py:169]:
```suggestion
    if not axes:
```

File: cupy/fft/fft.py (hunk 1 above)
Comment (leofang) [reply cupy/fft/fft.py:425]:
```suggestion
    if not axes_sorted:
```

**Comment by leofang** [comment]
Thanks, @toslunar. That's a good suggestion on PEP8. I just fixed it. PTAL.

**Comment by toslunar** [comment]
LGTM

Jenkins, test this please.

**Comment by toslunar** [comment]
The cherry-pick conflicted with #2746, which is only in `master` (v8).  I'm checking if the issue exists in `v7`.

**Comment by toslunar** [comment]
`v7` raises `IndexError: tuple index out of range` or `IndexError: list index out of range`.

**Pull request merged**